### ISSUE TITLE
[Question/Feedback]:  AMBA Services - Azure Policy definitions feedback & bugs

### ISSUE THREAD

**Issue by riosengineer** [Azure/azure-monitor-baseline-alerts#394]
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Description

Hi AMBA team :)

This is following on from my last issue: https://github.com/Azure/azure-monitor-baseline-alerts/issues/383 and it also follows on from: https://github.com/Azure/azure-monitor-baseline-alerts/issues/364 

For those of us who want to leverage `services` definitions in other ways outside of `bicep build` and loading individual definitions into `loadDefinitions` as outlined [here](https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/deploy/Introduction-to-deploying-the-ALZ-Pattern/#customizing-the-amba-policies), such as via Enterprise Policy as Code, we cannot currently. This is because (most, if not all?) definitions I have checked in the [services area](https://azure.github.io/azure-monitor-baseline-alerts/services/ApiManagement/service/) all seem to not be valid definitions for Azure Policy.

My example: https://azure.github.io/azure-monitor-baseline-alerts/services/ApiManagement/service/ `UnauthorizedRequests - Metric Alert`

I go to load this into Azure Policy but get this error:
```
 The inner exception 'Evaluation result of language expression '[[parameters('MonitorDisableTagValues')]' is type 'String', expected type is 'Array'.'.
```

To fix:

change from (removing double square brackets):

```
          {
            "field": "[[concat('tags[', parameters('MonitorDisableTagName'), ']')]",
            "notIn": "[[parameters('MonitorDisableTagValues')]"
          }
```

to
```
          {
            "field": "[concat('tags[', parameters('MonitorDisableTagName'), ']')]",
            "notIn": "[parameters('MonitorDisableTagValues')]"
          }
```

then I am presented with another error `"The policy '1eea700e-f050-4eef-8517-78852495f79c' has defined parameters 'severity,windowSize,evaluationFrequency,autoMitigate,enabled,threshold,effect,MonitorDisableTagName' which are not used in the policy rule. Please either remove these parameters from the definition or ensure that they are used in the policy rule."`

To fix: remove all double square brackets in the `policyRule` block. Now I can successfully load the definition into Azure Policy via the Portal. 

I understand the idea of the pattern is, if you want custom definitions that you add them into the `loadDefinitions` and do a `bicep build`. But at the very least I would expect:

- all policy definitions on the https://azure.github.io/azure-monitor-baseline-alerts/services/ to be valid - can this be fixed?
- It seems even if I were to manually fix the definitions like above, they are not setup to work with AMBA out the box. I would have to modify them further, e.g. SQL CPU Percent Alert definition fails remediation due to `Couldn't find a metric named master-app_cpu_percent.` 

If AMBA does not have any `policySet` initiatives for us to leverage on a per `service` basis, then at least, allow the community to take definitions and create our own initiatives to consume how we see fit (e.g. via EPAC). Otherwise, I am not sure what the point of those definitions are, unless I'm mistaken on their purpose?

Maybe thoughts from @jonnedev would be good on this too, given the issue they also rasied previously (now closed) is a similar ask.

**Comment by judyer28** [comment]
@riosengineer, thank you for your feedback and interest in AMBA.  I agree that the policies under the services area of AMBA need to be updated to provide alternative ways to deploy other than including in a Bicep build.  I am tagging this as an enhancement for now.  We will investigate how AMBA can better handle deployment of policies for alerts when deploying for individual service alerts outside of patterns such as ALZ.

**Comment by judyer28** [comment]
@riosengineer, an update was recently made to AMBA to cleanup the policy template code found in the Azure Resources section.  You should now be able to copy and paste the policy directly into the Azure portal.  There is also now a "Deploy to Azure" button for the alert policies.  Let me know if this fixes the issue.

**Comment by riosengineer** [comment]
> @riosengineer, an update was recently made to AMBA to cleanup the policy template code found in the Azure Resources section.  You should now be able to copy and paste the policy directly into the Azure portal.  There is also now a "Deploy to Azure" button for the alert policies.  Let me know if this fixes the issue.    

That's great! Thank you team! I'll try and grab some time to check it out 🚀

**Comment by riosengineer** [comment]
> [@riosengineer](https://github.com/riosengineer), an update was recently made to AMBA to cleanup the policy template code found in the Azure Resources section. You should now be able to copy and paste the policy directly into the Azure portal. There is also now a "Deploy to Azure" button for the alert policies. Let me know if this fixes the issue.

Hey @judyer28 I can confirm, I am now able to deploy policies from the services area, thanks for this and to everyone who did this work.

I should have enough to create my own initiatives based on AMBAs services (picking and choosing for our org requirements on a per service basis) and then export into Enterprise Policy as Code.

**Comment by judyer28** [comment]
Thanks for the feedback @riosengineer.  I will go ahead and close this one out.  May I also close out https://github.com/Azure/azure-monitor-baseline-alerts/issues/383 as it seems related?

**Comment by riosengineer** [comment]
> Thanks for the feedback [@riosengineer](https://github.com/riosengineer). I will go ahead and close this one out. May I also close out [#383](https://github.com/Azure/azure-monitor-baseline-alerts/issues/383) as it seems related?

Sure you can close them. Would be good to eventually get a way to create an initiative from the services per resource as a roadmap item though, IMO!

**Comment by judyer28** [comment]
I agree!  We will be looking into that as a potential future enhancement.
